src/Esfa.Vacancy.Infrastructure/InfrastructureRegistry.cs

```diff
@@ -14,35 +16,59 @@ namespace Esfa.Vacancy.Infrastructure
 {
     public sealed class InfrastructureRegistry : StructureMap.Registry
     {
+        private readonly IProvideSettings _provideSettings;
+
         public InfrastructureRegistry()
+            : this(new AppConfigSettingsProvider(new MachineSettings()))
+        { }
+
+        public InfrastructureRegistry(IProvideSettings provideSettings)
         {
+            _provideSettings = provideSettings;
+
             For<ILog>().Use(x => new NLogLogger(
                 x.ParentType,
                 x.GetInstance<IRequestContext>(),
                 GetProperties())).AlwaysUnique();
-            For<IProvideSettings>().Use(c => new AppConfigSettingsProvider(new MachineSettings()));
+            For<IProvideSettings>().Use(c => _provideSettings);
             For<IElasticClient>().Use(context => context.GetInstance<ElasticClientFactory>().GetClient());
             For<ICacheService>().Singleton().Use<AzureRedisCacheService>();
 
             For<IFrameworkCodeRepository>().Use<CachedFrameworkCodeRepository>()
                 .Ctor<IFrameworkCodeRepository>().Is<FrameworkCodeRepository>();
-            
+
             For<IStandardRepository>().Use<CachedStandardRepository>()
                 .Ctor<IStandardRepository>().Is<StandardRepository>();
+
+            RegisterCreateApprenticeshipService();
         }
 
         private IDictionary<string, object> GetProperties()
         {
-            var properties = new Dictionary<string, object>();
-            properties.Add("Version", GetVersion());
+            var properties = new Dictionary<string, object> { { "Version", GetVersion() } };
             return properties;
         }
 
-        private string GetVersion()
+        private static string GetVersion()
         {
             var assembly = Assembly.GetExecutingAssembly();
             var fileVersionInfo = FileVersionInfo.GetVersionInfo(assembly.Location);
             return fileVersionInfo.ProductVersion;
         }
+
+        private void RegisterCreateApprenticeshipService()
+        {
+            var isRunningUnderSandboxEnvironment =
+                _provideSettings.GetNullableSetting(ApplicationSettingKeys.UseSandboxServices);
+
+            if (string.IsNullOrWhiteSpace(isRunningUnderSandboxEnvironment) == false)
```

Comment:
Worth switching the elses? if (string.IsNullOrWhiteSpace(isRunningUnderSandboxEnvironment)). Took me a while to get the double negative

Reply:
fixed